modules/serverlist.js

```diff
@@ -0,0 +1,81 @@
+// Made by epsilon#3892
+// https://forum.thd.vg/members/16800/
+
+var config = require("../config.json");
+var request = require("request")
+
+module.exports = function(client) {
+	var module = {};
+
+	var channel, message;
+	channel = client.channels.get(config.server_list.channel);
+	if (!channel) return;
+	channel.fetchMessage(config.server_list.message).then(msg => {
+		message = msg;
+		loop();
+	}).catch(err => {
+		return; // Don't do anything if the message doesn't exist
+	});
+
+	function loop() {
+		request({
+		    url: 'https://api.kag2d.com/v1/game/thd/kag/servers?filters=[{"field":"current","op":"eq","value":"true"},{"field":"connectable","op":"eq","value":true},{"field":"currentPlayers","op":"gt","value":"0"}]',
+		    json: true
+		}, function (error, response, body) {
+			if (error) return;
+
+			let servers = body.serverList;
+
+			// Sort servers and players
+			servers = servers.sort((a, b) => {
+				if (a.currentPlayers === b.currentPlayers) {
+					return a.name.toUpperCase().localeCompare(b.name.toUpperCase());
+				}
+				return b.currentPlayers - a.currentPlayers;
+			});
+			servers.forEach(server => {
+				server.playerList.sort((a, b) => {
+					return a.toUpperCase().localeCompare(b.toUpperCase());
+				});
+			});
+
+			// Make message content
+			let players = servers.reduce((t, x) => t + x.currentPlayers, 0);
+			let text = '```md\n' + `# Server list - ${servers.length} ${plural(servers.length, 'server')}, ${players} ${plural(players, 'player')}` + '``````diff\n';
+			text += servers.map(server => {
+				let prefix = (/(?=^KAG Official \w+ (AUS|AU|EU|US|USA)\b)|(?=^Official Modded Server (AUS|AU|EU|US|USA)\b)/g.test(server.name)) ? '-' : '+';
```

Comment:
shouldn't you check official by IP?

Reply:
it should but that requires it to be updated for every new official server added. regex hasn't failed me yet, and if it does, ill just make it more strict